MSCXXXX: Tokens for Contacting Accounts

thomasfiala opened the pull request:
---
name: WIP Proposal
about: A proposal that isn't quite ready for formal review yet.
title: '[WIP] Your Proposal Title'
labels: proposal
assignees: ''

---

<!-- Put your "rendered" link here -->

### Pull Request Checklist

<!-- Please read CONTRIBUTING.md before submitting your pull request -->

* [ ] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec-proposals/blob/master/CONTRIBUTING.md#sign-off)
* [ ] Update the title and file name of your proposal to match this PR's number (after opening).
* [ ] Pull request includes a ['Rendered' link](https://matrix.org/docs/spec/proposals#process) above.
* [ ] Have a look at the [MSC Checklist](MSC_CHECKLIST.md) for guidelines on various aspects of your MSC.

<!-- Once the proposal is ready for review, ask in [#matrix-spec:matrix.org](https://matrix.to/#/#matrix-spec:matrix.org) to get it marked as such. -->

turt2live commented on proposals/4264-token-for-contacting-accounts.md:
Implementation requirements:
* [ ] Client
* [ ] Server

Johennes commented on proposals/4264-token-for-contacting-accounts.md:
```diff
@@ -0,0 +1,106 @@
+## Proposal
+
+Baiscally, it's something like plus-addressing for e-mail, just with requiring
+and managing a token.
```
I found the use of "token" slightly confusing when first reading this. [RFC5233](https://datatracker.ietf.org/doc/html/rfc5233) calls the subaddressing part in email addresses "detail". I'm not sure if this is much better though. 🤔

thomasfiala replied:
I added this concern in the 'Alternatives' section. I'm also not sure whether 'detail' is less confusing, but I'm happy to change it to 'detail' or any better wording.

Johennes commented on proposals/4264-token-for-contacting-accounts.md:
```diff
@@ -0,0 +1,106 @@
+* There should be no automatic feedback to the requester whether the token is
+  valid or not, to avoid brute-force attacks.
+
+## Potential issues
```
User IDs are [limited](https://spec.matrix.org/v1.13/appendices/#user-identifiers) to 255 bytes. So depending on the length of your ID, your options for defining sub IDs could be limited.

thomasfiala replied:
I added this comment to the description. This issue might be solved by not treating the token to be part of the user ID but rather a separate part (with again e.g. 255 bytes).

Johennes commented on proposals/4264-token-for-contacting-accounts.md:
```diff
@@ -0,0 +1,106 @@
+memorizable phrases like `connect`, or random ASCII stings.
+
+Thus, for initiating a first contact, both persistent information of the account
+(user `alice` and server `matrix.org`) and a flexible token is required. This
```
It might be good to list ways in which senders could obtain the token. It seems that it either has to be shared outside of Matrix or via (public) room membership?

thomasfiala replied:
I added some examples, I hope this helps.

Johennes commented on proposals/4264-token-for-contacting-accounts.md:
```diff
@@ -0,0 +1,106 @@
+landing.
+
+
+## Alternatives
```
Some of the block- and allow-list filtering methods for invites in https://github.com/matrix-org/matrix-spec-proposals/pull/4192 might be alternatives.

thomasfiala replied:
I added a reference to [4192]. Does it make sense to state a back-reference there as well?

Johennes replied:
Oh, yes! I have added it. Feel free to resolve this comment and the other ones above which you've addressed.

thomasfiala commented on proposals/4264-token-for-contacting-accounts.md:
```diff
@@ -0,0 +1,106 @@
+(user `alice` and server `matrix.org`) and a flexible token is required. This
+has several benefits:
+
+* Users that only know `@alice:matrix.org` cannot contact Alice.
```
Both. I added a part to be more clear on what I believe to be a good split between server and client(s).

spaetz commented:
Hi there, great ideas. Given that direct chats are noting but specific rooms, I am wondering if this MSC should not be more generic in the sense that e.g. "allowing tokens to enter a room" could solve this issue just as well and provide a more generic solution.
Let's say I have a matrix room for a University course that I teach, I could provide a "token" (essentially a password) to those students enrolled in my course. (or think zoom session passwords). It would be more liberal than invitations, but less open than "public". Is this MSC really only useful for contacting individuals?

thomasfiala commented:
> Hi there, great ideas. Given that direct chats are noting but specific rooms, I am wondering if this MSC should not be more generic in the sense that e.g. "allowing tokens to enter a room" could solve this issue just as well and provide a more generic solution. Let's say I have a matrix room for a University course that I teach, I could provide a "token" (essentially a password) to those students enrolled in my course. (or think zoom session passwords). It would be more liberal than invitations, but less open than "public". Is this MSC really only useful for contacting individuals?

Thank you @spaetz. I implemented your proposal into the text. Accordingly, I also updated the title. I hope that it's not confusing, though. Any idea to rephrase it is highly welcome. Possibly the idea to establish semi-public rooms is more generic and requires a separate proposal?

turt2live commented on proposals/4264-token-for-contacting-accounts.md:
Overall: Based on https://github.com/matrix-org/matrix-spec-proposals/pull/4192, I feel as though this MSC would be best positioned as an extension to https://github.com/matrix-org/matrix-spec-proposals/pull/4155 rather than being a dedicated mechanism (for the parts that aren't already covered).

thomasfiala replied:
Yes, you can definitively see it this way. I'm open to grouping/organizing this MSC in any way that makes more sense.
Just one thing to keep in mind: #4155 currently appears to be a more or less "standard" whitelist/blacklist feature. The concept of "tokens" and, subsequently, the meas of managing and displaying them, might possibly create confusion if being mixed up with #4155.
